pkg/kev/config/extensions.go

```diff
@@ -138,18 +138,17 @@ func K8SCfgFromCompose(svc *composego.ServiceConfig) (K8SConfiguration, error) {
 }
 
 func WorkloadRestartPolicyFromCompose(svc *composego.ServiceConfig) string {
-	if svc.Deploy == nil || svc.Deploy.RestartPolicy == nil {
-		return ""
+	policy := RestartPolicyAlways
+
+	if svc.Restart != "" {
+		policy = svc.Restart
 	}
 
-	switch svc.Deploy.RestartPolicy.Condition {
-	case "on-failure":
-		return RestartPolicyOnFailure
-	case "none":
-		return RestartPolicyNever
-	default:
-		return RestartPolicyAlways
+	if svc.Deploy != nil && svc.Deploy.RestartPolicy != nil {
+		policy = svc.Deploy.RestartPolicy.Condition
```

Comment:
We shouldn't be returning a literal value from compose here. Previously we were mapping that value to the expected restart policy but this code was removed above.. As it stands this func will be returning mixed values.

Reply:
Unless this func is meant to return the literal value extracted from compose (if present)? In which case it'd be the extracted value or an empty string, so no priming policy with `RestartPolicyAlways` needed. Does the switch statement need to be moved somewhere else?

Reply:
This is the bit where we're moving validation to a different place. The default value used to be handled by project_service but now that doesn't make much sense since this configuration should be responsible for that part. The code as it preserves the functionality since the tests are still covering the same behaviour